### PR TITLE
#55/profile

### LINKED ISSUE #55: `ProfilePage.tsx` 🎛
### Expected Behavior

`ProfilePage.tsx` UI needs to be updated to match the figma

Image

### Current Behavior

We just have a placeholder there

### Any other notes

_No response_

### Assignment

This issue is free for anyone to take

### PR DESCRIPTION
**Pull request by camelPhonso**
# Description

**Closes #55**

Creates the profile page with user settings form.

The form originally displays the user information saved in our context but it does not currently update it - all changes are lost on refresh.

This pr needs the changes made to the backend ([here](https://github.com/enBloc-org/arebyte-plugin-backend/pull/18#issue-2585768226)) as the form will be expecting the user type to include two new fields.

### Files changed

- `components/ProfilePage/` - overalled the styling of the page and added the new form
- `BurgerMenu.css` - added `z-index` settings to avoid absolutely placed elements from showing above the burger menu when opened
- `components/ToggleSwitch` - new component
- `globals.css`- added new global styling classes for `.flex-column` and `.align-start`
- `userQuery.ts`, `store.ts` & `userTypes.ts` - adjusted types and query string to reflect the new fields in the `user` content type

### UI changes

Screenshot 2024-10-14 at 12 44 20

Screenshot 2024-10-14 at 12 44 32

### Changes to Documentation

n/a

# Tests

No new tests as this is mostly a UI pr. All previous tests are passing

### CODE REVIEW

File: src/components/page-components/ProfilePage/ProfilePage.css [hunk 1]
```diff
@@ -0,0 +1,115 @@
+.profile-page {
+  animation: slideRight 0.5s ease-out;
+}
+
+.profile-page--user-details {
+  grid-column: 1/3;
+  height: fit-content;
+  min-height: 5.5rem;
+  max-height: 414px;
+  /* display: flex;
```
Comment (nichgalzin) [inline src/components/page-components/ProfilePage/ProfilePage.css]: Remove commented out styles

File: src/store/store.ts [hunk 2]
```diff
@@ -31,13 +31,17 @@ interface Actions {
...
+  updateIsQuiet: (newStatus: boolean) => void
+  updateIsPaused: (newStatus: boolean) => void
 }
 
 const initialState: State = {
   user: {
     id: undefined,
     username: undefined,
     email: undefined,
+    birth_date: "01:01:1973",
```
Comment (nichgalzin) [inline src/store/store.ts]: Should we have default values? I mean I like the ones you chose. But maybe undefined is better?

**Review by nichgalzin — CHANGES_REQUESTED** [review]

File: src/store/store.ts (hunk 2 above)
Comment (camelPhonso) [reply src/store/store.ts]: yep, set these to undefined now

**Comment by camelPhonso** [comment]
Updated the backend in a [new pr](https://github.com/enBloc-org/arebyte-plugin-backend/pull/20#issue-2589324201) and pushed the requested changes here 👍

**Review by nichgalzin — APPROVED** [review]
Get in!

**Pull request merged**